File: CS/BarryBollin/BarryBollin/GameStates/GameState.cs

using Acacia_Builder;
using Audrey;
using Microsoft.Xna.Framework;
using Microsoft.Xna.Framework.Content;
using Microsoft.Xna.Framework.Graphics;
using System;


public abstract class GameState {
    protected GameStateManager gsm;
    protected SpriteBatch spriteBatch;
    protected Texture2D cursorSpr;

    protected Engine e1 = new Engine();
    protected Game1 game;

    public Entity creature1;

    protected RenderSystem rs;
    protected ClickableSystem cs;
    protected TextSystem ts;

    public ContentManager Content
{
    get;
    private set;
}
    

    protected GameState(Game1 game, GameStateManager gsm){
        this.gsm = gsm;
        this.game = game;
        Content = new ContentManager(game.Services);
        Content.RootDirectory = "Content";
        init();
    }
    public Texture2D loadTexture(string str)
    {
        return Content.Load<Texture2D>(str);
    }
    
    public virtual void init()
    {
        spriteBatch = new SpriteBatch(game.GraphicsDevice);

        rs = new RenderSystem(e1, spriteBatch);
        ts = new TextSystem(e1, spriteBatch);
        cs = new ClickableSystem(e1);
    }
    
    public virtual void update(GameTime gameTime)
    {
        handleInput();
        cs.update(gameTime);

        

    }
    public virtual void draw()
    {
        rs.draw();
        ts.draw();
    }
    public abstract void handleInput();
}

File: Java/SShooter/core/src/com/eddie/sshooter/GameStates/GameState.java
package com.eddie.sshooter.GameStates;

import com.eddie.sshooter.Entities.Bullet;
import com.eddie.sshooter.Entities.Player;
import com.eddie.sshooter.Main.MainGame;
import com.eddie.sshooter.Managers.GameStateManager;

import java.util.ArrayList;

public abstract class GameState {

    public Player player;
    protected GameStateManager gsm;
    protected GameState(GameStateManager gsm){
        this.gsm = gsm;
        init();
    }
    public abstract void init();
    public abstract void update(float dt);
    public abstract void draw();
    public abstract void handleInput();
    public abstract void dispose();
}
